test/quantization/test_marlin_qqq.py

```diff
@@ -0,0 +1,129 @@
+import torch
+import copy
+import pytest
+
+from torch import nn
+from torch.testing._internal.common_utils import TestCase, run_tests
+from torchao.utils import TORCH_VERSION_AT_LEAST_2_5
+from torchao.dtypes import MarlinQQQLayout
+from torchao.quantization.quant_api import (
+    quantize_,
+    int8_dynamic_activation_int4_weight,
+)
+from torchao.quantization.marlin_qqq import (
+    pack_to_marlin_qqq,
+    unpack_from_marlin_qqq,
+)
+from torchao.quantization.quant_primitives import (
+    choose_qparams_and_quantize_affine_qqq,
+    MappingType,
+)
+
+
+class MarlinQQQ(TestCase):
+    def setUp(self):
+        super().setUp()
+        torch.manual_seed(0)
+
+        self.input = torch.randn((32, 16, 4096), dtype=torch.float16, device="cuda")
+        self.model = (
+            nn.Sequential(
+                nn.Linear(4096, 21504),
```

Comment:
probably a little bigger than needed for testing

Reply:
finished